assets/src/edit-story/components/panels/stylePreset/presets.js

```diff
@@ -58,101 +57,76 @@ const presetCSS = css`
+  border-radius: 15px;
   overflow: hidden;
-  white-space: nowrap;
+  position: relative;
+  ${({ disabled }) => (disabled ? 'opacity: 0.4;' : '')}
 `;
 
-const HighLight = styled.span`
-  padding: 0 2px;
-  ${({ background }) => generatePatternStyles(background)}
-  box-decoration-break: clone;
+const Color = styled.button`
+  ${presetCSS}
+  ${({ color }) => generatePatternStyles(color)}
 `;
 
 function Presets({
   stylePresets,
   handleOnClick,
   isEditMode,
   isText,
-  textContent = 'Text',
+  isBackground,
 }) {
-  const { fillColors, textColors, textStyles } = stylePresets;
-
-  const getStylePresetText = (preset) => {
-    const isHighLight =
-      preset.backgroundTextMode === BACKGROUND_TEXT_MODE.HIGHLIGHT;
-    return (
-      <TextWrapper>
-        {isHighLight ? (
-          <HighLight background={preset.backgroundColor}>
-            {textContent}
-          </HighLight>
-        ) : (
-          textContent
-        )}
-      </TextWrapper>
-    );
-  };
+  const { fillColors, textColors } = stylePresets;
 
   const colorPresets = isText ? textColors : fillColors;
-  const hasColorPresets = colorPresets.length > 0;
-  const hasPresets = textStyles.length > 0 || hasColorPresets;
+  const hasPresets = colorPresets.length > 0;
 
   const colorPresetRenderer = (color, i, activeIndex) => {
+    const disabled = isBackground && !isEditMode && presetHasOpacity(color);
+    // @todo The correct text here should be: Page background colors can not have an opacity.
+    // However, due to bug with Tooltips/Popup, the text flows out of the screen.
+    const title = disabled
+      ? __('Opacity not allowed for Page', 'web-stories')
+      : null;
     return (
-      <Color
-        tabIndex={activeIndex === i ? 0 : -1}
-        color={color}
-        onClick={(e) => {
-          e.preventDefault();
-          handleOnClick(color);
-        }}
-        aria-label={
-          isEditMode
-            ? __('Delete color preset', 'web-stories')
-            : __('Apply color preset', 'web-stories')
-        }
-      >
-        {isEditMode && <Remove />}
-      </Color>
-    );
-  };
-
-  const stylePresetRenderer = (style, i, activeIndex) => {
-    return (
-      <Style
-        tabIndex={activeIndex === i || (!activeIndex && i === 0) ? 0 : -1}
-        styles={generatePresetStyle(style, true)}
-        onClick={(e) => {
-          e.preventDefault();
-          handleOnClick(style);
-        }}
-        aria-label={
-          isEditMode
-            ? __('Delete style preset', 'web-stories')
-            : __('Apply style preset', 'web-stories')
-        }
-      >
-        {getStylePresetText(style)}
-        {isEditMode && <Remove />}
-      </Style>
+      <WithTooltip title={title}>
+        <ColorWrapper disabled={disabled}>
+          <Transparent />
+          <Color
+            tabIndex={activeIndex === i ? 0 : -1}
+            color={color}
+            onClick={(e) => {
+              e.preventDefault();
```

Comment:
Oh yeah, this should also be removed (and the button should just have `.attrs({ type: 'button' })` added to prevent it submitting the design panel form).